File: vseros/2014-15/okr/salary.py
a = int(input())
b = int(input())
c = int(input())
n = int(input())

q = max((a+1)//2*2, 2*b, 2*c) # новая зарплата Алексая после "минимального уравнивания"
m = q - a + q // 2 - b + q // 2 - c # минимальный премиальный фонд, которого хватит уравнять
if m > n:
    print(0)
else:
    print(q - a + (n - m) // 4 * 2)
    print(q // 2 - b + (n - m) // 4)
    print(q // 2 - c + (n - m) // 4)

File: vseros/2014-15/okr/salary.cpp
//Окружной этап всероссийской олимпиады школьников по информатике Москва, 14 декабря 2014 г.
//Задача 2. Заработная плата

#include <iostream>

using namespace std;

int main(int argc, char *argv[])
{
    int a, b, c, n;
    cin >> a >> b >> c >> n;

    int m = max(a + a % 2, max(2*b, 2*c));
    int lack = (m - a) + (m/2 - b) + (m/2 - c);

    if (n < lack)
    {
      cout << 0 << endl;
      return 0;
    }

    int k = (n - lack) / 4;
    cout << m - a + 2 * k << endl;
    cout << m/2 - b + k << endl;
    cout << m/2 - c + k << endl;

    return 0;
}
